float8_experimental/float8_dynamic_linear.py

```diff
@@ -63,7 +63,9 @@ def __init__(self, **super_kwargs):
         super().__init__(**super_kwargs)
 
     def forward(self, x):
-        x_fp8 = cast_to_float8_e4m3fn(x, self.forward_config)
+        x_fp8 = cast_to_float8_e4m3fn(
```

Comment:
this looks like static scaling, because the reader has to go inside the `cast_to_float8_e4m3fn` to understand that the scale can actually be calculated if it's not provided

IMO something more readable would be

```
def forward(self, x):
    if is_training:
      ... training logic, unchanged
    else:
      assert is_inference
      custom scaling for activation
      don't call any grad stuff since we are in inference
```